app/models/manageiq/providers/ansible_tower/shared/automation_manager/configuration_script.rb

```diff
@@ -1,36 +1,38 @@
-class ManageIQ::Providers::AnsibleTower::AutomationManager::ConfigurationScript
-  module ApiCreate
+module ManageIQ::Providers::AnsibleTower::Shared::AutomationManager::ConfigurationScript
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def provider_object(connection = nil)
+      (connection || connection_source.connect).api.job_templates.find(manager_ref)
+    end
+
     def create_in_provider(manager_id, params)
       manager = ExtManagementSystem.find(manager_id)
       job_template = manager.with_provider_connection do |connection|
         connection.api.job_templates.create!(params)
       end
-
-      # Get the record in our database
-      # TODO: This needs to be targeted refresh so it doesn't take too long
       EmsRefresh.queue_refresh(manager, nil, true) if !manager.missing_credentials? && manager.authentication_status_ok?
-
       find_by(:manager_id => manager.id, :manager_ref => job_template.id)
     end
 
     def create_in_provider_queue(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+
+      manager = ExtManagementSystem.find(manager_id)
```

Comment:
We need to do it twice because of the non-deterministic nature of computers. LOL